src/Microsoft.OData.Edm/Csdl/Parsing/SchemaJsonParser.cs

```diff
@@ -0,0 +1,1564 @@
+                        // Annotations for $OnDelete are prefixed with $OnDelete.
+                        string termName;
+                        propertyName = SeperateAnnotationName(propertyName, out termName);
+                        if (termName != null)
+                        {
+                            // termname with "@" and optional "#"
+                            if (AnnotationJsonParser.TryParseCsdlAnnotation(termName, propertyValue, context, out CsdlAnnotation annotation))
+                            {
+                                // annotation for the navigation property
+                                navPropertyAnnotations.Add(annotation);
+                                break;
+                            }
+                            else if (propertyName == "$OnDelete")
+                            {
+                                // annotation for OnDelte
+                                onDeleteAnnotations.Add(annotation);
+                                break;
+                            }
+                        }
+                        else
+                        {
+                            ReportUnexpectedElement(propertyValue, context);
+                        }
+
+                        break;
+                }
+            });
+
+            // If it's collection navigation property, let's construct the type name as Collection(entityType).
+            if (collection != null && collection.Value)
+            {
+                typeName = "Collection(" + typeName + ")";
+            }
+
+            // Annotations for $OnDelete are prefixed with $OnDelete. However, It's not supported in ODL yet.
+            if (csdlOnDelete != null && onDeleteAnnotations.Count > 0)
+            {
+                onDeleteAnnotations.ForEach(a => csdlOnDelete.AddAnnotation(a));
+            }
+
+            CsdlNavigationProperty csdlNavProperty = new CsdlNavigationProperty(name, typeName, nullable, parter,
+                containsTarget != null ? containsTarget.Value : false, // Absence of the member means false.
+                csdlOnDelete,
+                referentialConstraints != null ? referentialConstraints : Enumerable.Empty<CsdlReferentialConstraint>(),
+                context.Location());
+
+            navPropertyAnnotations.ForEach(a => csdlNavProperty.AddAnnotation(a));
+            return csdlNavProperty;
+        }
+
+        public static CsdlOnDelete ParseCsdlOnDelete(JsonElement element, JsonParserContext context)
+        {
+            string onDelete = element.ParseAsString(context);
+
+            EdmOnDeleteAction edmOnDelete = EdmOnDeleteAction.None;
+            switch (onDelete)
+            {
+                case "Cascade":
+                    edmOnDelete = EdmOnDeleteAction.Cascade;
+                    break;
+                default:
+                    // So far, ODL only supports "Cascade", for others "SetNull, or SetDefault.", we use "None" for them

```

Comment:
>// So far, ODL only supports "Cascade", for others "SetNull, or SetDefault.", we use "None" for them [](start = 20, length = 100)

this seems like an easy thing to fix. can you open an issue (perhaps against V8, if we're concerned about a breaking change)

Reply:
Done at: [https://github.com/OData/odata.net/issues/1819](https://github.com/OData/odata.net/issues/1819)

---
In reply to: [443078936](https://github.com/OData/odata.net/pull/1775#discussion_r443078936) [](ancestors = 443078936)